### PR TITLE
[Compute] BREAKING CHANGE: `az vm/vmss create`: New VM from existing disk or image

### PR DESCRIPTION
**Pull request by ZengTaoxu**
**Related command**

**Description**

Close: #23476

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/vm/_constants.py [hunk 1]
```diff
@@ -11,8 +11,7 @@
 # --------------------------------------------------------------------------
 
 UPGRADE_SECURITY_HINT = 'Consider upgrading security for your workloads using Azure Trusted Launch VMs. ' \
-                        'To know more about Trusted Launch, please visit ' \
-                        'https://learn.microsoft.com/en-us/azure/virtual-machines/trusted-launch.'
+                        'To know more about Trusted Launch, please visit please visit https://aka.ms/TrustedLaunch'
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/vm/_constants.py:14]:
```suggestion
                        'To know more about Trusted Launch, please visit https://aka.ms/TrustedLaunch'
```

File: src/azure-cli/azure/cli/command_modules/vm/_validators.py [hunk 2]
```diff
@@ -1331,51 +1331,30 @@ def _validate_generation_version_and_trusted_launch(cmd, namespace):
...
-            compute_client = _compute_client_factory(cmd.cli_ctx, subscription_id=image_info[0])
-            gallery_image_info = compute_client.gallery_images.get(
-                resource_group_name=image_info[1], gallery_name=image_info[2], gallery_image_name=image_info[3])
-            generation_version = gallery_image_info.hyper_v_generation if hasattr(gallery_image_info,
-                                                                                  'hyper_v_generation') else None
-            features = gallery_image_info.features if hasattr(gallery_image_info, 'features') else None
-            trusted_launch_warning_log(namespace, generation_version, features)
-            return
+            if namespace.security_type is None:
+                namespace.security_type = 'Standard'
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/vm/_validators.py:1339]: May I ask why make this logic for image version id?

File: src/azure-cli/azure/cli/command_modules/vm/_validators.py [hunk 3]
```diff
@@ -1401,10 +1380,9 @@ def _validate_generation_version_and_trusted_launch(cmd, namespace):
         client = _compute_client_factory(cmd.cli_ctx).disks
         attach_os_disk_name = parse_resource_id(namespace.attach_os_disk)['name']
         attach_os_disk_info = client.get(namespace.resource_group_name, attach_os_disk_name)
-        generation_version = attach_os_disk_info.hyper_v_generation if hasattr(attach_os_disk_info,
-                                                                               'hyper_v_generation') else None
-        features = attach_os_disk_info.features if hasattr(attach_os_disk_info, 'features') else None
-        trusted_launch_warning_log(namespace, generation_version, features)
+        disk_security_profile = attach_os_disk_info.security_profile if hasattr(attach_os_disk_info,
+                                                                                'security_profile') else None
+        validate_vm_disk_trusted_launch(namespace, disk_security_profile)
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/vm/_validators.py:1385]: May I ask why make this logic for this scenario?

**Comment by yanzhudd** [comment]
Replaced by https://github.com/Azure/azure-cli/pull/27749

**Pull request closed without merging**